### PR TITLE
Startup script for cables

### PR DESCRIPTION
**Pull request by kr3ator**
Related Issue: n/a

## New Behavior

This PR adds new startup script for adding cables.

## Contrast to Current Behavior

n/a

## Discussion: Benefits and Drawbacks

This script supports all connection options. If one of the ends of the cable is a Circuit (CircuitTermination to be precise) it is being searched in a _get_or_create_ fashion.
Current drawback is that we don't have a script for Provider Networks. If there is a need for a cable which terminates at ProviderNetwork then this object must exist.

## Changes to the Wiki

...

## Proposed Release Note Entry

Feature: add startup script for cables

## Double Check

* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

### CODE REVIEW

**Comment by kr3ator** [comment]
The build pipelines will fail until we merge #742 - as I added there relevant PSUs to server04

File: initializers/cables.yml
Comment (cimnine) [inline initializers/cables.yml]:
```suggestion
##   termination_X_class - required if different than 'Interface', which is the default
```

File: initializers/cables.yml
Comment (cimnine) [inline initializers/cables.yml]: Perhaps you can add a link for the user where he/she can learn of new classes. But if that gets too complicated, then leave it be as it is.

File: initializers/cables.yml
Comment (cimnine) [inline initializers/cables.yml]:
```suggestion
## Required parameters for a circuit termination:
## 
## ```
## termination_x_circuit:
##   term_side - termination side of a circuit. Must be A or B
##   cid - circuit ID value
##   site OR provider_network - name of Site or ProviderNetwork respectively. If both provided, Site takes precedence
## ```
##
```

File: initializers/cables.yml
Comment (cimnine) [inline initializers/cables.yml]:
```suggestion
## Required parameters for termination X ('a' or 'b'):
##
## ```
## termination_X_name - name of interface
## termination_X_device - name of the device interface belongs to
## termination_X_class - required if different than Interface which is the default
## ```
## 
## Supported termination classes: Interface, ConsolePort, ConsoleServerPort, FrontPort, RearPort
```

This makes it clearer IMO which part is example code and which is text.

File: initializers/dcim_interfaces.yml
Comment (cimnine) [inline initializers/dcim_interfaces.yml:17]: Is `virtual` not supported anylonger?

Otherwise I suggest to leave one interface being `virtual`. (Or add an extra one that is `virtual`.)

File: startup_scripts/460_cables.py
Comment (cimnine) [inline startup_scripts/460_cables.py]: I'm wondering whether this list is realy necessary. It's just something to keep in sync with upstream's code, and I don't like that. Can you think of another way to do the check? Or can we ignore the check and catch the error later, so that the user still gets a decent error message?

File: startup_scripts/460_cables.py
Comment (cimnine) [inline startup_scripts/460_cables.py:225]: I believe we check in other scripts whether the resource was actually created or not, and only then show the message, don't we? If so, could you make this script behave the same?

**Review by cimnine — CHANGES_REQUESTED** [review]
Wow, that's much work! Thanks for the PR. I hope it was at least some fun doing this!

I did not go through each and every line of code, just the important bits (at least I hope I catched most of the imporant bits). But that lgtm, besides the remarks I've left. Feel free to oppose with your reasoning to any of my remarks when you disagree.

File: startup_scripts/460_cables.py
Comment (kr3ator) [reply startup_scripts/460_cables.py:225]: With Cables the _get_or_create()_ method doesn't do any good. In this script if we didn't crash until line 231 then it will be created. If a resource is not created there will be an exception raised.

File: startup_scripts/460_cables.py
Comment (kr3ator) [reply startup_scripts/460_cables.py]: I don't like this either ;-) I could just delete this as the same classes are already being imported at the top.

File: initializers/dcim_interfaces.yml
Comment (kr3ator) [reply initializers/dcim_interfaces.yml:17]: True, sorry I took over all the ports on that server. I'll add an extra one as virtual

File: initializers/cables.yml
Comment (kr3ator) [reply initializers/cables.yml]: OK

Comment (kr3ator) [reply initializers/cables.yml]: There is no single place where this is listed clearly.

**Comment by kr3ator** [comment]
> ... I hope it was at least some fun doing this!

99% was fun :-)

**Comment by kr3ator** [comment]
@tobiasge this is now ready

**Comment by kr3ator** [comment]
@cimnine @tobiasge is there something blocking the merge I should fix from your perspective?

File: initializers/device_types.yml
Comment (cimnine) [inline initializers/device_types.yml]: Why did you comment this section? (I mean the extra comment level of course)

**Review by cimnine — APPROVED** [review]

File: initializers/device_types.yml
Comment (kr3ator) [reply initializers/device_types.yml]: Sorry for that. It's a mistake done when I switched comments in what I've added to '# #' instead of '##'. Will fix this now

**Comment by kr3ator** [comment]
@tobiasge can we merge this?

**Comment by tobiasge** [comment]
@kr3ator Would you be willing to update the cable initializers for the Netbox 3.3 release?

**Comment by kr3ator** [comment]
@tobiasge sorry I just noticed your message. I can do it if that's not done already.

**Pull request merged**
